### PR TITLE
Replace 'API Key' with 'Project Token'

### LINKED ISSUE #25: `api_key` should be `project_token`
With the expansion of the API we did earlier this year, we consolidated our nomenclature on:

- tokens not keys
- API Token for user's tokens, used for API access
- Project Token for authentication of metric-sending agents

Given that, we should at least change the user-facing bits to be `project_token` instead of `api_key`. Ideally, we change all the internal stuff too.

### PR DESCRIPTION
**Pull request by janxious**
Fixes #25 

We should use consistent language when we talk about our project tokens/api tokens/api keys/authentication shards/🎱.

TODO: update each of the various installation techniques where 'api key' exists and then test them.

- [X] windows installer (REMOVED, N/A)
- [x] puppet template
- [x] chef recipes
- [x] packages built by fpm (rpms and whatnot)

### CODE REVIEW

**Comment by tonydewan** [comment]
Proposal: rename it "authentication shards" instead. :D

**Comment by janxious** [comment]
This is currently kind of mixed up with a bunch of other stuff, namely, packaging, installation, installation testing, etc. So tangly!

**Comment by mediocretes** [comment]
Plans for tomorrow's test party:

- delete all of the kitchen instances / clear out virtual box
- verify all of the chef/puppet installers
- check all of the /etc/instrumetnald.toml files for project token (vs api key)
- merge this, make remove win32 its own thing

**Comment by mediocretes** [comment]
I think the goals of this PR have been met, and then some.

File: Rakefile [hunk 1]
```diff
@@ -64,17 +64,17 @@ set BUNDLE_IGNORE_CONFIG=
...
+                              runtime:      TRAVELING_RUBY_FILE % "linux-x86",
+                              arch:         "i386",
+                              packages:     %w{deb rpm},
+                              platform:     "linux",
+                              packagecloud: true,
+                              wrapper:      WRAPPER_SCRIPT_SHELL,
+                              separator:    '/',
+                              package_from_compressed: true,
+                              dest_dir:     DEST_DIR
+                             },
```
Comment (tonydewan) [inline Rakefile]: Extra spaces?

File: bin/instrumentald [hunk 2]
```diff
@@ -140,17 +140,17 @@ command = ARGV.first && ARGV.first.to_sym
...
 def show_option_error(option_parser, message)
   STDERR.puts "ERROR: #{message}\n\n"
   print option_parser.help
   exit 1
 end
 
-if options[:api_key].to_s.strip.empty? && !File.exist?(options[:config_file])
-  show_option_error option_parser, "You must provide an API key, or a config path that exists(#{options[:config_file]})"
-end
+# if options[:project_token].to_s.strip.empty? && !File.exist?(options[:config_file])
```
Comment (tonydewan) [inline bin/instrumentald]: Why is this commented out?

File: chef/instrumentald/templates/default/instrumentald.toml.erb [hunk 3]
```diff
@@ -1,5 +1,11 @@
-# Replace YOUR_API_KEY with the API key for your Instrumental project and
+# Replace the project_token attribute with the project token from an Instrumental project and
 # remove the # from the beginning of the following line to have instrumentald
-# start sending metrics to your account.
+# start sending metrics for your project.
```
Comment (tonydewan) [inline chef/instrumentald/templates/default/instrumentald.toml.erb]: This is kind of a weird place for a newline...

**Comment by tonydewan** [comment]
Looking good to me! :+1:

**Pull request merged**